quaternion.py

```diff
@@ -258,10 +258,10 @@ def from_euler(roll, pitch, yaw):
         st = np.sin(pitch / 2.0)
         ss = np.sin(yaw / 2.0)
 
-        return Quaternion(np.array([[cp * ct * cs - sp * st * ss],
-                                    [sp * st * cs + cp * ct * ss],
-                                    [sp * ct * cs + cp * st * ss],
-                                    [cp * st * cs - sp * ct * ss]]))
+        return Quaternion(np.array([[cp * ct * cs + sp * st * ss],
+                                    [sp * ct * cs - cp * st * ss],
+                                    [cp * st * cs + sp * ct * ss],
+                                    [cp * ct * ss - sp * st * cs]]))
```

Comment:
Could you tell me why this change was necessary? 

It's not that I think you're wrong, I'm just relatively certain that I had used this.

Also, could you add a test that converts in and out of Euler?

Reply:
Yeah, when creating a quaternion using the original code from Euler angles of pure roll, pitch, or yaw, the resulting quaternion's vector didn't match the rotation. Pure roll resulted in a quaternion rotating about the y axis, pure pitch about the z axis, and pure yaw about the x axis. 

To verify the changes I made, I looked in your `geometry` library and found that `pytransform` didn't match. The changes I made now match `geometry`. 

I can add some tests.

Reply:
I added a test for the `from_euler` function and got rid of the import changes I made. So it should work with Python2, but the imports don't work correctly for Python3